lib/games/games_module.rb

```diff
@@ -17,19 +16,44 @@ def add_game
     print 'Please state how many years ago it was published: '
     published_date = gets.chomp
     new_game = Game.new(title, multiplayer, last_played_at, published_date)
-    @game_list.push(new_game)
+    @game_list.push(title: new_game.title, multiplayer: new_game.multiplayer, last_played_at: new_game.last_played_at, published_date: new_game.published_date)
     puts 'Game created succesfully'
     puts ''
   end
 
   def list_all_games
-    @game_list.each do |game|
-      puts "Title: \"#{game.title}\", Multiplayer: #{game.multiplayer}, Last Time Played:#{game.last_played_at} years ago, Published #{game.published_date} years ago"
+    @game_list = JSON.parse(File.read('./lib/games/json_data/games.json'))
+    if @game_list.empty?
+      puts 'There are no games in the list yet, please add one first'
+    else
+      @game_list.each do |game|
+        puts "Title: #{game['title']}, Multiplayer: #{game['multiplayer']}, Last Time Played:#{game['last_played_at']} years ago, Published #{game['published_date']} years ago"
+      end
     end
   end
 
   def list_all_authors
-    puts 'There are no authors in the list yet, please add one first'
+    @authors_list = JSON.parse(File.read('./lib/games/json_data/authors.json'))
+    if @authors_list.empty?
+      puts 'There are no games in the list yet, please add one first'
+    else
+      @authors_list.each do |author|
+        puts "Title: #{author['first_name']}, Multiplayer: #{author['last_name']}"
+      end
+    end
+  end
+
+  def read_games_data
+    if File.exist?('./lib/games/json_data/games.json')
+      File.open('./lib/games/json_data/games.json', 'a') { |f| f.write(JSON.generate([])) }
+    elsif File.exist?('./lib/games/json_data/authors.json')
+      File.open('./lib/games/json_data/authors.json', 'a') { |f| f.write(JSON.generate([])) }
+    end
```

Comment:
When a File exists this method is unable to read the game correctly.

Here's what I get when saving state with the save_games_data.

File: './lib/games/json_data/games.json' : 
  `[{"title":"Zelda Ocarina of Time","multiplayer":false,"last_played_at":"2","published_date":"10"}]`
 
I would expect that this JSON content would be loaded to the app class, but currently, it's not. The read_games_data is using a flag that writes and appends to the end of that file, and then it's converting a [] to JSON. I suggest modifying this function because it will cause an exception if the file already exists. If you check line 25 you can imagine that the parse method will not work anymore, since the games.json file will be like this:

  `[{"title":"Zelda Ocarina of Time","multiplayer":false,"last_played_at":"2","published_date":"10"}][]` => this last [] will raise an exception in JSON.parse method.

Reply:
Everything was a bug I introduced because I changed the name of the folder, but now the bug it's fixed, thanks for this great review. Very professional 👍 😎